Minor pathfinder-related changes (preparation for mobile refactor)

reaperrr opened the pull request:
This is a bunch of presumably "safe", easy to review changes from #7565.

Rydra committed:
Changes to improve understandability of code

reaperrr committed:
Minor pathfinder changes

Mailaender commented:
Looks good. :+1: Can't notice any changes in-game. :white_check_mark:

pchote commented on OpenRA.Game/Actor.cs:
```diff
@@ -172,6 +172,11 @@ public void QueueActivity(Activity nextActivity)
 				currentActivity.Queue(nextActivity);
 		}
 
+		public bool IsAlliedWith(Actor actor)
```
I'm not convinced about this.  Alliances are defined in terms of players, not actors, and this just adds a duplicate code path.

reaperrr commented:
Updated. Travis error is not my fault.
Removed `IsAlliedWith` and additionally removed an `as World` and `as Actor` from PathFinder which I had previously overlooked.

penev92 commented on OpenRA.Mods.Common/Pathfinder/PathGraph.cs:
```diff
@@ -127,17 +127,17 @@ int GetCostToNode(CPos destNode, CVec direction)
 			}
 
-			return InvalidNode;
+			return Constants.InvalidNode;
```
If you are replacing the `InvalidNode` from this class with `Constants.InvalidNode`, why are you not removing the old one?

reaperrr replied:
I think the original refactor PR forgot about that too (or I just overlooked it). Anyway, fixed.

obrakmann commented:
:+1:

reaperrr commented:
Huh? Does github auto-rebase now? Because this definitely had some changes overlapping with #8622.
